src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp

```diff
@@ -173,6 +202,91 @@ namespace xdp {
     //}
   }
 
+  void AIEProfilingPlugin::printTileModStats(xaiefal::XAieDev* aieDevice, const tile_type& tile, bool isCore)
+  {
+    auto col = tile.col;
+    auto row = tile.row + 1;
+    auto loc = XAie_TileLoc(col, row);
+    std::string modType = isCore ? "Core" : "Memory";
+    XAie_ModuleType mod = isCore ? XAIE_CORE_MOD :  XAIE_MEM_MOD;
+    std::stringstream msg;
+
+    const std::string groups[3] = {
+      XAIEDEV_DEFAULT_GROUP_GENERIC,
+      XAIEDEV_DEFAULT_GROUP_STATIC,
+      XAIEDEV_DEFAULT_GROUP_AVAIL
+    };
+
+    msg << "Resource usage stats for Tile : (" << col << "," << row << ") Module : " << modType << std::endl;
+    for (auto&g : groups) {
+      auto stats = aieDevice->getRscStat(g);
+      auto pc = stats.getNumRsc(loc, mod, XAIE_PERFCNT_RSC);
+      auto ts = stats.getNumRsc(loc, mod, xaiefal::XAIE_TRACE_EVENTS_RSC);
+      auto bc = stats.getNumRsc(loc, mod, XAIE_BCAST_CHANNEL_RSC);
+      msg << "Resource Group : " << std::left <<  std::setw(10) << g << " "
+          << "Performance Counters : " << pc << " "
+          << "Trace Slots : " << ts << " "
+          << "Broadcast Channels : " << bc << " "
+          << std::endl;
+    }
+
+    xrt_core::message::send(severity_level::info, "XRT", msg.str());
+  }
+
+  uint32_t AIEProfilingPlugin::getNumFreeCtr(xaiefal::XAieDev* aieDevice,
+                                          const std::vector<tile_type>& tiles,
+                                          bool isCore,
+                                          const std::string& metricSet)
+  {
+    uint32_t numFreeCtr = 0;
+    uint32_t tileId = 0;
+    XAie_ModuleType mod = isCore ? XAIE_CORE_MOD : XAIE_MEM_MOD ;
+    auto stats = aieDevice->getRscStat(XAIEDEV_DEFAULT_GROUP_AVAIL);
+
+    for (unsigned int i=0; i < tiles.size(); i++) {
+      auto loc = XAie_TileLoc(tiles[i].col, tiles[i].row + 1);
+      auto avail = stats.getNumRsc(loc, mod, XAIE_PERFCNT_RSC);
+      if (i == 0) {
+        numFreeCtr = avail;
+      } else {
+        if (avail < numFreeCtr) {
+          numFreeCtr = avail;
+          tileId = i;
+        }
+      }
+    }
+
+    auto& requestedEvents = isCore ? mCoreStartEvents[metricSet] : mMemoryStartEvents[metricSet];
+    auto& EventStrings =    isCore ? mCoreEventStrings[metricSet] : mMemoryEventStrings[metricSet];
+
+    auto numTotalEvents = requestedEvents.size();
+    if (numFreeCtr < numTotalEvents) {
+      std::stringstream msg;
+      std::string modType = isCore ? "core" : "memory";
+      msg << "Only " << numFreeCtr << " out of " << numTotalEvents
+          << " metrics were available for aie "
+          << modType <<  " module profiling due to resource constraints. "
+          << "AIE profiling uses performance counters which could be taken up by aie trace, ecc etc."
```

Comment:
Please change "taken up by aie trace, ecc etc." to "already used by AIE trace, ECC, etc." or alike.